Bottom: c0365c4529c68459be0ffa226714e412efafda03
Top:    c3d50a80a1bfe3ce3e8cc3c8997ab130f782afde
Author: Robert Collins <robertc@robertcollins.net>
Date:   2013-03-08 19:44:51 +1300

Refresh of define-stream-result

---

diff --git a/doc/for-framework-folk.rst b/doc/for-framework-folk.rst
index 8408a91..514c4b7 100644
--- a/doc/for-framework-folk.rst
+++ b/doc/for-framework-folk.rst
@@ -150,16 +150,23 @@ The design has several key principles:
 * Make the API as simple as possible - each aspect should do one thing well.
 
 The ``TestResult`` API this is intended to replace has three different clients.
-The testrunner running tests uses the API to find out whether the test run had
-errors, how many tests ran and so on. Secondly each executing ``TestCase``
-notifies the ``TestResult`` about activity. Finally, each ``TestCase`` queries
-the ``TestResult`` to see whether the test run should be aborted.
+
+* Each executing ``TestCase`` notifies the ``TestResult`` about activity.
+
+* The testrunner running tests uses the API to find out whether the test run
+  had errors, how many tests ran and so on.
+
+* Finally, each ``TestCase`` queries the ``TestResult`` to see whether the test
+  run should be aborted.
 
 With ``StreamResult`` we need to be able to provide a ``TestResult`` compatible
-adapter, to allow incremental migration. However, we don't need to conflate
-things long term. So - we define three separate APIs. Firstly there is the
-``StreamResult`` API which handles events generated by running tests. See the
-API documentation for ``testtools.StreamResult`` for details.
+adapter (``StreamToExtendedDecorator``) to allow incremental migration.
+However, we don't need to conflate things long term. So - we define three
+separate APIs, and merely mix them together to provide the
+``StreamToExtendedDecorator``. ``StreamResult`` is the first of these APIs -
+meeting the needs of ``TestCase`` clients. It handles events generated by
+running tests. See the API documentation for ``testtools.StreamResult`` for
+details.
 
 StreamSummary
 -------------
diff --git a/testtools/tests/test_testresult.py b/testtools/tests/test_testresult.py
index 397f058..0bb6a0a 100644
--- a/testtools/tests/test_testresult.py
+++ b/testtools/tests/test_testresult.py
@@ -458,6 +458,7 @@ class TestStreamResultContract(object):
         result.stopTestRun()
 
     def test_files(self):
+        # Test parameter combinations when files are being emitted.
         result = self._make_result()
         result.startTestRun()
         self.addCleanup(result.stopTestRun)
@@ -475,6 +476,7 @@ class TestStreamResultContract(object):
             result.status(file_name=_u("foo"), file_bytes=_b("bar"), **kwargs)
 
     def test_test_status(self):
+        # Tests non-file attachment parameter combinations.
         result = self._make_result()
         result.startTestRun()
         self.addCleanup(result.stopTestRun)
